src/Security/UserClassBuilder.php

```diff
@@ -97,10 +98,17 @@ private function addGetUsername(ClassSourceManipulator $manipulator, UserClassCo
             );
         }
 
+        $getterIdentifierName = 'getUsername';
+
+        // Check if we're using Symfony 5.3+ - UserInterface::getUsername() was replaced with UserInterface::getUserIdentifier()
+        if (class_exists(InMemoryUser::class)) {
```

Comment:
This class DOES only exist on 5.3. But why did you choose this class specifically? Could we use something more direct like `method_exists(Symfony\Component\Security\Core\User\User::class, 'getUserIdentifier')`?

Reply:
`User::class` has been deprecated as of 5.3 && Renamed / Will be renamed to `InMemoryUser::class`

https://github.com/symfony/symfony/blob/abeb8e4ca9adfcc1494df22af9fd9a1c0181b532/src/Symfony/Component/Security/Core/User/User.php#L21

https://github.com/symfony/symfony/pull/40443